File: services/language-service/src/main/java/dev/animedia/languageservice/application/service/CreateLanguageService.java
package dev.animedia.languageservice.application.service;

import dev.animedia.languageservice.application.dto.LanguageDto;
import dev.animedia.languageservice.application.mapper.LanguageApplicationMapper;
import dev.animedia.languageservice.application.usecase.CreateLanguageUseCase;
import dev.animedia.languageservice.domain.exception.LanguageCodeAlreadyExistsException;
import dev.animedia.languageservice.domain.model.Language;
import dev.animedia.languageservice.domain.repository.LanguageCommandRepository;
import dev.animedia.languageservice.domain.repository.LanguageQueryRepository;

public class CreateLanguageService implements CreateLanguageUseCase {
    private final LanguageCommandRepository languageCommandRepository;
    private final LanguageQueryRepository languageQueryRepository;
    private final LanguageApplicationMapper languageApplicationMapper;

    public CreateLanguageService(
        LanguageCommandRepository languageCommandRepository,
        LanguageQueryRepository languageQueryRepository,
        LanguageApplicationMapper languageApplicationMapper
    ) {
        this.languageCommandRepository = languageCommandRepository;
        this.languageQueryRepository = languageQueryRepository;
        this.languageApplicationMapper = languageApplicationMapper;
    }

    @Override
    public LanguageDto create(LanguageDto request) {
        if(languageQueryRepository.findByCode(request.code()).isPresent()) {
            throw new LanguageCodeAlreadyExistsException();
        }

        Language language = languageApplicationMapper.toDomain(request);

        Language saved = languageCommandRepository.create(language);

        return languageApplicationMapper.toDto(saved);
    }
}

File: services/social-service/src/core/Application/Language/Service/Private/CreateLanguageService.php
<?php

declare(strict_types=1);

namespace Core\Application\Language\Service\Private;

use Core\Application\Language\DTO\Private\LanguageDto;
use Core\Application\Language\Exception\LanguageExistsException;
use Core\Application\Language\Mapper\LanguageApplicationMapperInterface;
use Core\Application\Language\UseCase\Private\CreateLanguageUseCase;
use Core\Domain\Language\Repository\LanguageCommandRepositoryInterface;
use Core\Domain\Language\Repository\LanguageQueryRepositoryInterface;

final readonly class CreateLanguageService implements CreateLanguageUseCase
{
    public function __construct(
        private LanguageQueryRepositoryInterface $languageQueryRepository,
        private LanguageCommandRepositoryInterface $languageCommandRepository,
        private LanguageApplicationMapperInterface $languageApplicationMapper
    ) {}

    #[\Override]
    public function execute(LanguageDto $languageDto): LanguageDto
    {
        $exists = $this->languageQueryRepository->existsByIsoCode($languageDto->languageIsoCode);
        if ($exists) {
            throw new LanguageExistsException($languageDto->languageIsoCode);
        }

        $language = $this->languageApplicationMapper->toLanguage($languageDto);
        $created = $this->languageCommandRepository->create($language);

        return $this->languageApplicationMapper->toPrivateLanguageDto($created);
    }
}
